pmd-core/src/main/java/net/sourceforge/pmd/SourceCodeProcessor.java

```diff
@@ -203,8 +205,10 @@ private void processSource(Reader sourceCode, RuleSets ruleSets, RuleContext ctx
     private void determineLanguage(RuleContext ctx) {
         // If LanguageVersion of the source file is not known, make a
         // determination
-        if (ctx.getLanguageVersion() == null) {
-            LanguageVersion languageVersion = configuration.getLanguageVersionOfFile(ctx.getSourceCodeFilename());
+        LanguageVersion languageVersion = ctx.getLanguageVersion();
+        if (languageVersion == null) {
+            languageVersion = configuration.getForceLanguageVersion();
```

Comment:
Maybe we can write this a bit more explicit and introduce a local variable "forceLanguage". With `ctx.setLanguageVersion(languageVersion)` we either set the forceLanguage (if given) or we try to determine the language, which might be not clear on first sight.